packages/ics721/src/execute.rs

```diff
@@ -736,7 +759,47 @@ where
                                 }
                             },
                         ),
-                    ))
+                    );
+
+                // we migrate only in case CLASS_ID_AND_NFT_CONTRACT_INFO is not populated yet
```

Comment:
Maybe we can put this in a function so its clear that its legacy code to be deleted?

Reply:
moved to `migrate_legacy()`

Reply:
Thank you